File: apple/PersonaToolkit/Sources/ChatConfig.swift
import Foundation

/// Bootstraps an Orchestrator. The Backend instance is supplied here;
/// the orchestrator does not know how it was built. A wholly-local
/// config supplies an in-process Backend; a server-backed config
/// supplies an HTTP/SSE one. Same Orchestrator, same UI either way.
public protocol ChatConfig: Sendable {
    var conversationID: String { get }

    /// The Participant.id of the local user. Exactly one participant in
    /// `initialParticipants` must match.
    var localParticipantID: String { get }

    var initialParticipants: [any Participant] { get }

    var commands: [any Command] { get }
    var observingHooks: [any ObservingHook] { get }
    var gatingHooks: [any GatingHook] { get }
    var permissionStore: any PermissionStore { get }

    var backend: any Backend { get }
    var display: any DisplayConfig { get }
}

File: packages/chat/src/contract/configuration/ChatConfig.ts
import type { Backend } from '../backend/Backend'
import type { Command } from '../commands/Command'
import type { GatingHook } from '../hooks/GatingHook'
import type { ObservingHook } from '../hooks/ObservingHook'
import type { Participant } from '../participants/Participant'
import type { PermissionStore } from '../permissions/PermissionStore'
import type { DisplayConfig } from './DisplayConfig'

export interface ChatConfig {
  readonly conversationID: string
  readonly localParticipantID: string
  readonly initialParticipants: ReadonlyArray<Participant>
  readonly commands: ReadonlyArray<Command>
  readonly observingHooks: ReadonlyArray<ObservingHook>
  readonly gatingHooks: ReadonlyArray<GatingHook>
  readonly permissionStore: PermissionStore
  readonly backend: Backend
  readonly display: DisplayConfig
}
